airbyte-integrations/bases/base-normalization/normalization/transform_catalog/transform.py

```diff
@@ -139,6 +101,280 @@ def extract_schema(profiles_yml: dict) -> str:
+        else:
+            raise KeyError("json_schema is not defined for " + stream_name)
+        # Below commented code is to handle namespacing in issue #1921
+        # if "target_namespace" in configuredStream:
+        #     schema: str = configuredStream["target_namespace"]
+        # else:
+        #     raise KeyError("target_namespace is not defined in configuredStream for " + stream_name)
+        # if "alias_name" in configuredStream:
+        #     name: str = configuredStream["alias_name"]
+        # else:
+        #     raise KeyError("alias_name is not defined in configuredStream for " + stream_name)
+        name = stream_name
+
+        schema = normalize_identifier_name(schema, integration_type)
+        raw_schema = normalize_identifier_name("_airbyte_" + schema, integration_type)
+        # Below commented code is to handle namespacing in issue #1921
+        # table = jinja_call(f"source('{raw_schema}', '{normalize_identifier_name(name, integration_type)}')")
+        # if raw_schema not in source_tables:
+        #     source_tables[raw_schema] = set()
+        #     all_tables[schema] = set()
+        # if name not in source_tables[raw_schema]:
+        #     source_tables[raw_schema].add(name)
+        #     all_tables[schema].add(name)
+        raw_name = normalize_identifier_name(f"_airbyte_raw_{name}", integration_type)
+        table = jinja_call("source('{}', '{}')".format(schema, raw_name))
+        if schema not in source_tables:
+            source_tables[schema] = set()
+            all_tables[schema] = set()
+        if raw_name not in source_tables[schema]:
+            source_tables[schema].add(raw_name)
+            source_tables[schema].add(name)
+            all_tables[schema].add(raw_name)
+            all_tables[schema].add(name)
+        else:
+            raise KeyError(f"Duplicate table {name} in {schema}")
+
+        # Check properties
+        if not properties:
+            raise EOFError("Unexpected empty properties in catalog")
+
+        process_node(
+            output=output,
+            integration_type=integration_type,
+            path=[name],
+            json_col=f"'{json_col}'",
+            tables_in_schema=all_tables,
+            properties=properties,
+            raw_schema=raw_schema,
+            schema=schema,
+            name=name,
+            table=table,
+            parent_hash_id="",
+            inject_sql_prefix="",
+            inject_sql_suffix="",
+        )
+    return source_tables
+
+
+def process_node(
+    output: str,
+    integration_type: str,
+    path: list,
+    json_col: str,
+    tables_in_schema: Dict[str, Set[str]],
+    properties,
+    raw_schema,
+    schema,
+    name,
+    table,
+    parent_hash_id: str,
+    inject_sql_prefix: str,
+    inject_sql_suffix: str,
+):
+    # Check properties
+    if not properties:
+        print("Ignoring '{}' nested field from {} because properties list is empty".format(name, "/".join(path)))
```

Comment:
is there a reason not to do `f"Ignoring {name} ... from {'/'.join(path)} because.."` ? I find that format to be more readable, but it's definitely a nitpick

Reply:
I can't use `{{ }}` as literal characters inside the `f"blabla"` strings and since this is using a lot of `{{ }}` for jinja, it works best with `.format()` so I kept using that pattern...